packages/admin/cms-admin/src/common/getEntityRanges.ts

```diff
@@ -0,0 +1,92 @@
+import type { CharacterMetadata } from "draft-js";
+import type { List } from "immutable";
+import { is, OrderedSet } from "immutable";
+
+type EntityKey = string | undefined | null;
+type Style = OrderedSet<string>;
+type StyleRangeWithId = [string, { style: string; id: number }[]];
+type StyleRange = [string, Style];
+type EntityRange = [EntityKey, Array<StyleRangeWithId>];
+export type CharacterMetaList = List<CharacterMetadata>;
+
+export const EMPTY_SET: Style = OrderedSet();
+
+export default function getEntityRanges(text: string, charMetaList: CharacterMetaList): EntityRange[] {
```

Comment:
Did you copy this functions from somewhere? If yes, there should be a comment referencing the original source